docs/UserGuide.md

```diff
@@ -195,6 +195,28 @@ Examples:
 
 * `lesson delete 2` deletes the 2nd lesson in the lessons schedule.
 
+### Finding lessons : `lesson find`
+
+Finds lessons whose student's name contain any of the given keywords.
```

Comment:
I'm not sure if I like the phrasing of this sentence.
Maybe you can try, "
Find lessons with students whose names contain any of the specified keywords."